llama_index/response_synthesizers/tree_summarize.py

```diff
@@ -113,16 +124,24 @@ def get_response(
             else:
                 response = self._service_context.llm_predictor.predict(
                     summary_template,
+                    output_cls=self._output_cls,
                     context_str=text_chunks[0],
                 )
-            return response
+
+            # return pydantic object if output_cls is specified
+            return (
+                response
+                if self._output_cls is None
+                else self._output_cls.parse_raw(response)
+            )
```

Comment:
curious: why do we convert to and return json string at the llm predictor level, and then convert back to pydantic object at the synthesizer level here? Is it mostly due to typing constraints?